sdk/storage/azure-storage-blobs/src/blob_responses.cpp

```diff
@@ -58,4 +59,48 @@ namespace Azure { namespace Storage { namespace Blobs {
     };
   }
 
+  void ListBlobsPageResult::OnNextPage(const Azure::Core::Context& context)
+  {
+    auto contextRef = std::cref(context);
+    if (&context == &Azure::Core::Context::GetApplicationContext())
+    {
+      // Called via iterator increment, use the same context as the last call.
+      contextRef = std::cref(m_context);
+    }
+
+    _detail::BlobRestClient::BlobContainer::ListBlobsSinglePageOptions protocolLayerOptions;
+    protocolLayerOptions.Prefix = m_operationOptions.Prefix;
+    protocolLayerOptions.ContinuationToken = NextPageToken;
+    protocolLayerOptions.MaxResults = m_operationOptions.PageSizeHint;
+    protocolLayerOptions.Include = m_operationOptions.Include;
+    auto response = _detail::BlobRestClient::BlobContainer::ListBlobsSinglePage(
+        *m_blobContainerClient->m_pipeline,
+        m_blobContainerClient->m_blobContainerUrl,
+        protocolLayerOptions,
+        contextRef.get());
+    for (auto& i : response.Value.Items)
+    {
+      if (i.Details.AccessTier.HasValue() && !i.Details.IsAccessTierInferred.HasValue())
+      {
+        i.Details.IsAccessTierInferred = false;
+      }
+      if (i.VersionId.HasValue() && !i.IsCurrentVersion.HasValue())
+      {
+        i.IsCurrentVersion = false;
+      }
+      if (i.BlobType == Models::BlobType::AppendBlob && !i.Details.IsSealed)
+      {
+        i.Details.IsSealed = false;
+      }
+    }
```

Comment:
@JeffreyRichter we seem to already be iterating through the list of results returned from the service. Is that expected?

The `ListBlobsSinglePageResult` model type just contains a single `std::vector<BlobItem> Items;`

Reply:
This loop is iterating over the single collection to set some default values. It's not allocating any memory so I think this is fine.